### PR TITLE
Initial import of the tsrepl nextjs app

### PR DESCRIPTION
**Pull request by orta**
Re: #114 which replaces the entire front-end with a feature parity implementation of the Playground

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,3 +1,5 @@
+Multiple Licenses
```
Comment (gillchristian) [inline LICENSE:1]: I think we could use the same license for everything and add the clarification regarding the new client here

File: client/package.json [hunk 2]
```diff
@@ -1,56 +1,38 @@
...
-    "@types/node": "^12.0.0",
-    "@types/react": "^16.9.0",
-    "@types/react-dom": "^16.9.0",
-    "@types/webpack-env": "^1.15.2",
-    "eslint-config-prettier": "^6.10.1",
-    "eslint-config-typescript": "^3.0.0",
-    "eslint-plugin-prettier": "^3.1.3",
-    "prettier": "2.0.4"
+  "prettier": {
+    "printWidth": 140,
```
Comment (gillchristian) [inline client/package.json:35]: Chaotic-good, I like it! :rocket:

File: client/public/next.svg [hunk 3]
```diff
@@ -0,0 +1 @@
...55v12.6H80.5V0h68.7zm34.3 0h-17.8l63.8 79.4h17.9l-32-39.7 32-39.6h-17.9l-23 28.6-23-28.6zm18.3 56.7-9-11-27.1 33.7h17.8l18.3-22.7z"/><path fill="#000" d="M81 79.3 17 0H0v79.3h13.6V17l50.2 62.3H81Zm252.6-.4c-1 0-1.8-.4-2.5-1s-1.1-1.6-1.1-2.6.3-1.8 1-2.5 1.6-1 2.6-1 1.8.3 2.5 1a3.4 3.4 0 0 1 .6 4.3 3.7 3.7 0 0 1-3 1.8zm23.2-33.5h6v23.3c0 2.1-.4 4-1.3 5.5a9.1 9.1 0 0 1-3.8 3.5c-1.6.8-3.5 1.3-5.7 1.3-2 0-3.7-.4-5.3-1s-2.8-1.8-3.7-3.2c-.9-1.3-1.4-3-1.4-5h6c.1.8.3 1.6.7 2.2s1 1.2 1.6 1.5c.7.4 1.5.5 2.4.5 1 0 1.8-.2 2.4-.6a4 4 0 0 0 1.6-1.8c.3-.8.5-1.8.5-3V45.5zm30.9 9.1a4.4 4.4 0 0 0-2-3.3 7.5 7.5 0 0 0-4.3-1.1c-1.3 0-2.4.2-3.3.5-.9.4-1.6 1-2 1.6a3.5 3.5 0 0 0-.3 4c.3.5.7.9 1.3 1.2l1.8 1 2 .5 3.2.8c1.3.3 2.5.7 3.7 1.2a13 13 0 0 1 3.2 1.8 8.1 8.1 0 0 1 3 6.5c0 2-.5 3.7-1.5 5.1a10 10 0 0 1-4.4 3.5c-1.8.8-4.1 1.2-6.8 1.2-2.6 0-4.9-.4-6.8-1.2-2-.8-3.4-2-4.5-3.5a10 10 0 0 1-1.7-5.6h6a5 5 0 0 0 3.5 4.6c1 .4 2.2.6 3.4.6 1.3 0 2.5-.2 3.5-.6 1-.4 1.8-1 2.4-1.7a4 4 0 0 0 .8-2.4c0-.9-.2-1.6-.7-2.2a11 11 0 0 0-2.1-1.4l-3.2-1-3.8-1c-2.8-.7-5-1.7-6.6-3.2a7.2 7.2 0 0 1-2.4-5.7 8 8 0 0 1 1.7-5 10 10 0 0 1 4.3-3.5c2-.8 4-1.2 6.4-1.2 2.3 0 4.4.4 6.2 1.2 1.8.8 3.2 2 4.3 3.4 1 1.4 1.5 3 1.5 5h-5.8z"/></svg>
```
Comment (gillchristian) [inline client/public/next.svg:1]: I assume this file is not used, right?

**Review by gillchristian — APPROVED** [review]
Looking good! :rocket: 

I changed the base branch to `v2` so we can continue merging there until ready to go live.

Regarding the link shortener, we could make it into it's own tab on the right, rather than being a plugin.

For the future I'd like for it to support GitHub login and add some features like displaying visits of the links you've created and allowing to update what the short points to (now once the short is created it always points to the same code, which is a problem if something needs to be fixed in the code for example).

But I'll do that at some point once I get some time :grimacing:

File: client/public/next.svg (hunk 3 above)
Comment (orta) [reply client/public/next.svg:1]: No 👍🏻

**Comment by orta** [comment]
Note to myself: this sha of the playground doesn't seem to have gist support

I'd have expected: https://tsplay-git-tsrepl-gillchristian.vercel.app?#gist/927ccc66ae3022dc64c4f650109b937a-2

To work

**Comment by gillchristian** [comment]
It seems like it's not working on the official playground either

At least this doesn't work for me https://www.typescriptlang.org/play?#gist/927ccc66ae3022dc64c4f650109b937a-2

**Comment by orta** [comment]
No, I think the support was dropped sometime, I was noting that I need to get that back

**Comment by orta** [comment]
Got a sense of what we need to do to get this up and running? I guess I have write access so I can self merge, what happens if I do?

**Comment by gillchristian** [comment]
Hey @orta, sorry for the radio silence, I've been a bit all over the place lately. We can merge to the `v2` branch.

To release to production I'd like to have some sort of `/about` page and a native version of the shortener plugin. Anything else you'd like to have?

Here's the v2 branch running https://tsplay-git-v2-gillchristian.vercel.app/

**Pull request merged**